pack/nodejs/helm.ts

```diff
@@ -34,8 +34,9 @@ export namespace v2 {
     export class Chart extends pulumi.ComponentResource {
         public readonly resources: {[key: string]: pulumi.CustomResource};
 
-        constructor(releaseName: string, config: ChartOpts, opts?: pulumi.ResourceOptions) {
-            super("kubernetes:helm.sh/v2:Chart", releaseName, config, opts);
+        constructor(releaseName: string, config: ChartOpts, opts?: pulumi.CustomResourceOptions) {
```

Comment:
This is a `ComponentResource`, so should be `pulumi.ComponentResourceOptions`.

I'll push a change to fix this, along with a test.

Reply:
But the whole point is that I want to pass a provider that is used by the child resources .... Is the convention here different for components?

Reply:
Oh I think this may be the automatic parent provider searching stuff you guys mentioned last week. If it works, great, certainly was not obvious to me.

Reply:
Yes - all of this is related to how providers are handled in components (via parent).

You would do:

```javascript
let myk8s = new k8s.Provider("myk8s", { /* ... */ });
let chart = new helm.v2.Chart("simple-nginx", { /* ... */ }, { providers: { "kubernetes": myk8s } });
```

This is necessary, instead of `provider:` because components can create many resources from many providers.

Opened https://github.com/pulumi/docs/issues/516 to document this and other related concepts.